### PR TITLE
Discover plugin to fetch sources from dist-git

### PR DESCRIPTION
**Pull request by lukaszachy**
First step on #585

### CODE REVIEW

**Comment by lukaszachy** [comment]
@psss @jscotka Is this a valid direction?
Seems I will need to patch existing discover steps to allow skipping 'copy to workdir'.

**Comment by lukaszachy** [comment]
A lot of work ahead however seems to be doing something
```
$ tmt run discover -vvv  plans --name /sources
/var/tmp/tmt/run-167

/plans/sources/url
    discover
        how: sources
        order: 50
        how: fmf
        name: delegated from sources
        order: 50
        directory: /var/tmp/tmt/run-167/plans/sources/url/discover/default/tests
        summary: 56 tests selected
            /tests/core/adjust
            /tests/core/docs
            /tests/core/dry
            /tests/core/enabled
....

```

File: tmt/steps/discover/sources.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                rpmbuilddir,
+                os.listdir(rpmbuilddir)[0]
+            ),
+            testdir
+        )
+
+        # Run shell if it was there
+        shell = self.get('shell')
+        if shell:
+            self.run(shell, cwd=testdir)
```
Comment (pvalena) [inline tmt/steps/discover/sources.py:90]: So basically, you run any abitrary command on host.

This is what I wanted to avoid.

Comment (lukaszachy) [reply tmt/steps/discover/sources.py:90]: It is not required - just a hotfix because tmt sources doesn't contain .fmf/version.

Comment (lukaszachy) [reply tmt/steps/discover/sources.py:90]: And frankly  - we need `rpmbuild -bp` to run on host. If you really want to put some malicious code there you can / and will be able to.
This discover plugin is optional - you don't need to install it

Comment (pvalena) [reply tmt/steps/discover/sources.py:90]: You're right, I think it should use some method of isolation then.

Comment (lukaszachy) [reply tmt/steps/discover/sources.py:90]: What if it runs in mock?

**Comment by lukaszachy** [comment]
Note to self: https://pagure.io/standard-test-roles/blob/master/f/roles/standard-test-source/library/source-lookaside.py doesn't use *pkg binary, lets get inspired.

**Comment by lukaszachy** [comment]
Do we want to support fetching URL first and extract sources from there? E.g.
```
discover:
   how: fmf
   from: sources
   url: https://src.fedoraproject.org/rpms/tmt.git
   path: tmt
```

**Comment by psss** [comment]
I like the idea of explicitly mentioning `distgit` as the feature is limited to `distgit` usage only. What about using it as the prefix?

    distgit-source: true
    distgit-type: FedoraDistGit

Or, in order to make it a bit more user-friendly, use the distinct part of the class name only?

    distgit-source: true
    distgit-type: Fedora

Or should it be rather `dist-git-source` and `dist-git-type`? What do you think?

**Comment by lukaszachy** [comment]
> I like the idea of explicitly mentioning `distgit` as the feature is limited to `distgit` usage only. What about using it as the prefix?
> 
> ```
> distgit-source: true
> distgit-type: FedoraDistGit
> ```
> 
> Or, in order to make it a bit more user-friendly, use the distinct part of the class name only?
> 
> ```
> distgit-source: true
> distgit-type: Fedora
> ```
> 
> Or should it be rather `dist-git-source` and `dist-git-type`? What do you think?

I've found https://github.com/release-engineering/dist-git so `dist-git-source:true` and `dist-git-type: Fedora` (to make it less typing for users :) Thanks for the ideas.

**Comment by psss** [comment]
Tests failed with:

    bash: line 1: ./sources.sh: No such file or directory

Looks like a forgotten file?

**Comment by lukaszachy** [comment]
Renamed file to distigit . And it was easier to run the test directly (not using tmt) :/

**Comment by pvalena** [comment]
LGTM.

**Review by psss — APPROVED** [review]
Thanks a lot for implementing this, @lukaszachy. Looks good in general. I just ran into problem with wrong option names which caused the feature would not work when enabled from a plan. This should now be fixed by cccc94e. I've also extended the test a bit and updated the spec to make it structured (the list of available options was already quite long). Please, have a look and check if everything's ok.

**Review by lukaszachy — COMMENTED** [review]
LGTM, thanks for the adjustments @psss

**Pull request merged**